### PR TITLE
MGDAPI-5887 - Create go module for 3scale apis to import to integreatly-opeartor

### PR DESCRIPTION
**Pull request by valerymo**
# Issue link
https://issues.redhat.com/browse/MGDAPI-5887

### This PR will be merged after 3scale PR: https://github.com/3scale/3scale-operator/pull/854
**PR require update of go.mod - replace private repo reference to 3scale-operator**: `github.com/valerymo/3scale-operator v0.0.0-20230831081915-2a8db134a385`

# What
Reduced the number of 3-scale dependencies in Roam.
This was achieved through a 3scale refactoring and reducing APIs dependencies on PKG packages in 3scale. The result is the same as we checked for API module, so final decision was Not to create separate apis module.

# Verification steps
- get current branch
- check the number of dependencies - count of files in vendors dir, compare with master:
```
$ pwd
.../integreatly-operator/vendor
[vendor] (master)$ find . -type f |wc -l
7001
[vendor] $ find . -type f |wc -l
6770
$ date
Thu Aug 31 12:49:27 IDT 2023
```
- Install Rhoam locally, and check that installation is successfull
```
[integreatly-operator] (MGDAPI-5887)$ oc get rhmi rhoam -n redhat-rhoam-operator -o json | jq -r '.status.stage'
complete
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -161,3 +156,21 @@ replace k8s.io/client-go => k8s.io/client-go v0.26.3
 
 // Required until the following jira is resolved - https://issues.redhat.com/browse/MGDAPI-4996
 replace github.com/grafana-operator/grafana-operator/v4 => github.com/grafana-operator/grafana-operator/v4 v4.2.0
+
+//replace github.com/3scale/3scale-operator/apis => ../3scale-operator/apis
+
+replace github.com/3scale/3scale-operator/apis => github.com/valerymo/3scale-operator/apis v0.0.0-20230809091710-a1fe409c3f23
+
+replace (
```
Comment (MStokluska) [inline go.mod:164]: Hey,
Is this going to stay like that? I think we should not be making that many replaces, in general using replaces in RHOAM can lead to a lot of issues later on.

File: go.mod [hunk 2]
```diff
@@ -157,3 +150,6 @@ replace github.com/openshift/api => github.com/openshift/api v0.0.0-202108310919
 
 // Required until the following jira is resolved - https://issues.redhat.com/browse/MGDAPI-4996
 replace github.com/grafana-operator/grafana-operator/v4 => github.com/grafana-operator/grafana-operator/v4 v4.2.0
+
+// Required until 3scale PR merge https://github.com/3scale/3scale-operator/pull/854
+replace github.com/3scale/3scale-operator => github.com/valerymo/3scale-operator v0.0.0-20230831101242-4e5f852570a0
```
Comment (austincunningham) [inline go.mod:155]: Think we can remove this replace. Should only be in your local branch not in the PR

File: vendor/modules.txt [hunk 3]
```diff
@@ -1445,3 +1417,4 @@ sigs.k8s.io/structured-merge-diff/v4/value
 sigs.k8s.io/yaml
 # github.com/openshift/api => github.com/openshift/api v0.0.0-20210831091943-07e756545ac1
 # github.com/grafana-operator/grafana-operator/v4 => github.com/grafana-operator/grafana-operator/v4 v4.2.0
+# github.com/3scale/3scale-operator => github.com/valerymo/3scale-operator v0.0.0-20230831101242-4e5f852570a0
```
Comment (austincunningham) [inline vendor/modules.txt:1420]: And here

File: go.mod (hunk 2 above)
Comment (austincunningham) [reply go.mod:155]: As the 3scale pr is merged we can definitely remove this.

File: pkg/products/threescale/reconciler.go [hunk 4]
```diff
@@ -52,7 +52,8 @@ import (
 	userHelper "github.com/integr8ly/integreatly-operator/pkg/resources/user"
 
 	threescalev1 "github.com/3scale/3scale-operator/apis/apps/v1alpha1"
-	threescaleAmp "github.com/3scale/3scale-operator/pkg/3scale/amp/component"
+	//threescaleAmp "github.com/3scale/3scale-operator/pkg/3scale/amp/component"
```
Comment (carlkyrillos) [inline pkg/products/threescale/reconciler.go:55]: @valerymo Can this commented-out import for `threescaleAmp` be removed?

File: go.mod (hunk 2 above)
Comment (valerymo) [reply go.mod]: done. Thank you

File: pkg/products/threescale/reconciler.go (hunk 4 above)
Comment (valerymo) [reply pkg/products/threescale/reconciler.go]: Done. Thank you for review @carlkyrillos

**Comment by valerymo** [comment]
@austincunningham , @MStokluska , @carlkyrillos  I addressed comments, maybe PR could progress? (if you will have time for this). Thank you

**Comment by austincunningham** [comment]
:eyes:

**Comment by austincunningham** [comment]
@valerymo confirmed that verification steps out of date , no need to check for 3scale-operator/apis module in go.mod

/lgtm

**Comment by valerymo** [comment]
Validation notes - updated, removed sentence regarding 3scale-operator/apis module in go.mod
It's outdated. We tried to add separate module for 3scale-operator/apis on early stages of PR development, but finally we found that separate module is not needed. Same effect could be achieved by removing dependencies in 3scale-operator `apis` from `pkg`. Thank you  for review and the comments @austincunningham

**Pull request merged**